src/server/repositories/UrlRepository.ts

```diff
@@ -93,15 +93,16 @@ export class UrlRepository implements UrlRepositoryInterface {
             })
           }),
         )
-        await Promise.all(
-          tagCreationResponses.map(async (response) => {
-            const [tag, _] = response
-            if (tag) {
-              // @ts-ignore, addTag is provided by Sequelize during run time.
-              await url.addTag(tag, { transaction: t })
-            }
-          }),
-        )
+        const tags: TagType[] = []
+        tagCreationResponses.forEach((response) => {
+          const [tag, _] = response
+          if (tag) {
+            tags.push(tag)
+          }
+        })
```

Comment:
Just wondering - is there ever a scenario in which `if (tag)` returns false? My understanding is if there was any tag that failed to be created in `Tag.findOrCreate`, then the whole transaction rolls back. 

If so, we can simplify the above code to:

```
        const tags: TagType[] = tagCreationResponses.map(([tag, _]) => tag)
```

Reply:
There are some validations at the DB side, so imagine we have another service calling the upsertTags without having the input validated by Joi object, the function will return empty tag object

Reply:
ok sounds good